src/pal/src/file/file.cpp

```diff
@@ -3605,18 +3700,30 @@ GetTempFileNameW(
             goto done;
         }
     }
-       
+    
+    tempfile_name = new char[MAX_LONGPATH+1];
+    if (tempfile_name == NULL)
+    {
+        pThread->SetLastError(ERROR_NOT_ENOUGH_MEMORY);
+        uRet = 0;
+        goto done;
+    }
+    
     uRet = GetTempFileNameA(full_name, 
                             (lpPrefixString == NULL) ? NULL : prefix_string,
                             0, tempfile_name);
-        
-    if ( uRet && !MultiByteToWideChar( CP_ACP, 0, tempfile_name, -1, 
-                                       lpTempFileName, MAX_PATH_FNAME ))
+                        
+    path_size = MultiByteToWideChar( CP_ACP, 0, tempfile_name, -1, 
```

Comment:
If the GetTempFileNameA fails, the tempfile_name can contain garbage. So you will need to keep the check of the uRet before calling the MultiByteToWideChar.